### PR TITLE
vdk-meta-jobs: add DAG with args example

### PR DESCRIPTION
**Pull request by yonitoo**
What:
Add a bit more complex VDK example for DAGs including the latest functionality - job arguments.
Additionally, concurrent running jobs limit is configured.

Signed-off-by: Yoan Salambashev <ysalambashev@vmware.com>

### CODE REVIEW

File: examples/dag-with-args-example/README.md [hunk 1]
```diff
@@ -0,0 +1,635 @@
+Overview
```
Comment (tozka) [inline examples/dag-with-args-example/README.md:1]: Please follow the https://github.com/vmware/versatile-data-kit/wiki/Tutorial-Guidelines

Comment (yonitoo) [reply examples/dag-with-args-example/README.md:1]: Thanks for the quick feedback! I based the structure of this tutorial on the other DAGs example. However, I will check the provided link and see what can be improved.

File: examples/dag-with-args-example/dag-job/dag_job.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from vdk.plugin.meta_jobs.meta_job_runner import MetaJobInput
+
+
+JOBS_RUN_ORDER = [
```
Comment (murphp15) [inline examples/dag-with-args-example/dag-job/dag_job.py:6]: I should probably have brought this up in one of the earlier dag PRs. 
But it is very weird that they are supposed to create list of dicts.
They are expected to know all the field names?
Can we not make this dict a class?
the class would be called DagNode.



Maybe this isn't the best place to bring this up. 

WDYT?

Comment (yonitoo) [reply examples/dag-with-args-example/dag-job/dag_job.py:6]: You have a point here. This data modelling idea was brought up a while ago. There was even a task to do it but it was dropped primarily due to user feedback. Kinda surprisingly, the users turned out to be happy with the list of dicts. Therefore, we left it that way for now.

File: examples/dag-with-args-example/README.md [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+Also, make sure to install quickstart-vdk in a separate Python virtual environment.
+
+This example also requires Trino DB installed.
+See the Trino [Official Documentation](https://trino.io/) for more details about installation.
+
+Please note that this example requires deploying Data Jobs in a Kubernetes
+environment, which means that you would also need to install
+the **VDK Control Service.**
+
+<ins>Prerequisites</ins>:
```
[the remark this answers is not in the document]
Comment (yonitoo) [reply examples/dag-with-args-example/README.md:64]: several lines above this there is the "pip install quickstart-vdk" and a link to [Installation page](https://github.com/vmware/versatile-data-kit/wiki/Installation#install-sdk ). However, if you think that it should also be added to this Prerequisites list, I can add bullet point "Install [quickstart-vdk](https://github.com/vmware/versatile-data-kit/wiki/Installation#install-quickstart-vdk)".

[the remark this answers is not in the document]
Comment (yonitoo) [reply examples/dag-with-args-example/README.md:64]: I can actually attach a link to the quickstart-vdk guide here: "Also, make sure to [install quickstart-vdk](https://github.com/vmware/versatile-data-kit/wiki/Installation#install-quickstart-vdk) ...."

File: examples/dag-with-args-example/README.md (hunk 1 above)
Comment (yonitoo) [reply examples/dag-with-args-example/README.md]: Adapted the example to the tutorial guidelines.

**Review by murphp15 — APPROVED** [review]
Looks good!

File: examples/dag-with-args-example/dag-job/dag_job.py (hunk 2 above)
Comment (murphp15) [reply examples/dag-with-args-example/dag-job/dag_job.py:6]: If the customer wants it this way then I'm happy to mark it as resolved.

**Pull request merged**